plugins/event_notification/providers/kafka/lib/model/KafkaNotificationTemplate.php

```diff
@@ -0,0 +1,323 @@
+	 * @throws AvroSchemaParseException
+	 * @throws \FlixTech\SchemaRegistryApi\Exception\SchemaRegistryException
+	 * @throws kCoreException
+	 */
+	private function getAvroPayload($topicName, $msg)
+	{
+		list($schema, $schemaId) = $this->getSchemaInfo($topicName);
+		if(!($schema && $schemaId))
+		{
+			return null;
+		}
+		return $this->buildAvroPayload($schemaId, $schema, $msg);
+	}
+	
+	private function getSchemaInfo($subject)
+	{
+		$schemaId = null;
+		$schema = null;
+		
+		if(!kConf::hasMap('schemaRegistry'))
+		{
+			throw new kCoreException("schema registry configuration file (schemaRegistry.ini) wasn't found!");
+		}
+		
+		$schemaRegistryConfig = kConf::getMap('schemaRegistry');
+		
+		$schemaRegistryServer = isset($schemaRegistryConfig['schema_registry_server']) ? $schemaRegistryConfig['schema_registry_server'] : null;
+		$schemaRegistryPort = isset($schemaRegistryConfig['schema_registry_port']) ? $schemaRegistryConfig['schema_registry_port'] : null;
+		if(!($schemaRegistryServer && $schemaRegistryPort))
+		{
+			KalturaLog::debug("Message configured with Avro payload but cannot find schema registry settings!");
+			return array(null, null);
+		}
+		
+		$currentSchemaVersion = $schemaRegistryConfig[$subject];
+		if(!$currentSchemaVersion)
+		{
+			KalturaLog::debug("Missing schema version for [$subject]!");
+			return array(null, null);
+		}
+		
+		$key = $subject . '_' . $currentSchemaVersion;
+		
+		$cache = kCacheManager::getSingleLayerCache(kCacheManager::CACHE_TYPE_AVRO_SCHEMAS);
+		if($cache)
+		{
+			$schemaInfo = $cache->get($key);
+			if($schemaInfo)
+			{
+				$schemaId = $schemaInfo[self::SCHEMA_ID];
+				$schema = $schemaInfo[self::SCHEMA];
+			}
+		}
+		
+		if(!$schema)
+		{
+			$schemaRegistry = new BlockingRegistry(
+				new PromisingRegistry(
+					new Client(['base_uri' => $schemaRegistryServer . ":" . $schemaRegistryPort])
+				)
+			);
+			
+			if(!$schemaRegistry)
+			{
+				return array(null, null);
+			}
+			
+			$schema = $schemaRegistry->schemaForSubjectAndVersion($subject, $currentSchemaVersion);
+			
+			if(!$schema)
+			{
+				KalturaLog::debug("Missing schema in schema registry for [$subject]!");
+				return array(null, null);
+			}
+			
+			$schemaId = $schemaRegistry->schemaId($subject, $schema);
+			if(!$schemaId)
+			{
+				KalturaLog::debug("Missing schema ID for subject [$subject] and schema [$schema]!");
+				return array(null, null);
+			}
+			
+			if($cache)
+			{
+				$schemaInfo = array(
+					self::SCHEMA_ID => $schemaId,
+					self::SCHEMA => $schema
+				);
+				
+				$result = $cache->add($key, $schemaInfo);
+				if(!$result)
+				{
+					KalturaLog::debug("Avro schema [$subject] did not save to cache [$key]");
+				}
+			}
+		}
+		
+		return array($schema, $schemaId);
+	}
+	
+	/**
+	 * @param int $schemaId
+	 * @param AvroSchema $schema
+	 * @param array $msg
+	 * @return string
+	 * @throws AvroIOException
+	 */
+	private function buildAvroPayload($schemaId, $schema, $msg)
```

Comment:
change all private functions to protected

Reply:
Done